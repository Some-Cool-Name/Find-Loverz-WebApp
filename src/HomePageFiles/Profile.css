div.profile{
    display: flex;
    width: fit-content;
    width: fit-content;
    height: fit-content;
    height:max-content;
    flex-wrap: nowrap;
    min-width: 700px;
}


.right-container-1 {
    min-width: 400px;
}

.profile-image {
    width: 120px;
    height: 110px;
    padding-top: 10px;
    margin: auto;
    margin-bottom: 10px;
    border-radius: 100px;
    background-image: url(../default.png);
    background-position: center;
    background-size: cover;
    background-repeat: no-repeat;
}
p.editProfile{
    color: blue;
    font-size: small;
    margin-bottom: 10px;
}
h2.usernameAge{
    color: rgb(160, 157, 157);
    font-weight: bolder;
    font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
    margin-bottom: 10px;
}
h2.usernameAge2{
    color: rgb(160, 157, 157);
    font-weight: bolder;
    font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
    margin-bottom: 10px;
    margin-left: 10px;
    text-align: start;
}
p.location{
    margin-bottom: 10px;
    margin-left: 10px;
    text-align: start;
    color: blue;
}

h2.txtAccInfo{
    text-align: start;
    font-size:medium ;
    padding-left: 5px;
}
hr.divider{
    width: 140px;
    background: linear-gradient(to right,#c471ed,#b621fe , #12c2e9);
}
dir{
    padding-left: 0;
}
P.lblDiscriber{
    text-align: start;
    font-weight: bolder;
    font-size: medium;
    padding-left:15PX ;
    
}
.wrapper{
    padding: 20px;
   
}
input.inputfield{
    border:none;
    border-bottom: 2px solid silver;
    border-radius: 0%;
    
    color: lightslategrey;
    font-size: smaller;
}
textarea.bio{
    border:none;
    resize: none;
    border-bottom: 2px solid silver;
    color: lightslategrey;
    padding: 10px;
    width: 320.333px;

}

textarea[type="text"]:focus{
    border:none;
    resize: none;
    outline: 0;
    border-bottom: 2px solid #00bafa;
    box-shadow:0 2px 0 0 #c471ed;   
}
div.interest-element{
    display:inline-block ;
    margin-top: 10px;
    
}
div.interest-element2{
    display:inline-block ;
    margin: 5px;
    
}
.interest-element {
    height: 18px;
    width: fit-content;
    border-radius:15px;
    background-color: white;
    border-style: solid;
    border-width: 1px;
}
.interest-element2 {
    width: 80px;
    height: 15px;
    border-radius:15px;
    background-color: white;
    border-style: solid;
    border-width: 1px;
    font-size: x-small;
}
div.interest-element:hover{
    background-color: white;
}
p.editInterest{
    margin-top: 10px;
    color: blue;
    font-size: small;
    margin-bottom: 10px;
   text-align: right;
   margin-left: 20px;
}
hr.line2{
    
    background: silver;
}
.btnUpdate{
    margin: auto;
    padding: auto; 
}
div.btnUpdate{
  margin-top: 20px;
  display:inline-block ; 
}
#register-pic{
    width:70px;
   clear: both;
    height: 70px;
    margin-bottom: 0px;
    padding-top: 0px;
    margin-top: 0px;
    border-radius: 100px;
    background-image: url(../default.png);
    background-position: center;
    background-size: cover;
    background-repeat: no-repeat;
}
input#fileupload::before {
    content: 'edit';
    display: inline-flex;
    border: none;
    border-radius: 50px;
    padding: 5px;
    font-weight: 700;
    font-size: 5px;
    text-transform: uppercase;
    color: #ffffff;
    background: linear-gradient(to right,#b621fe , #12c2e9);
    font-size: 14px;
    font-weight: 900;
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
    cursor: pointer;
    outline: none;
    white-space: nowrap;
    -webkit-user-select: none;
}
input#fileupload {
    width: 50px;
    margin: auto;
    height: max-content;
    border: none;
    border-radius: 10%;
    padding: 0;
    size: 2px;
}
a.button{
    margin: 0px 0px;
    margin-top: 0px;
}