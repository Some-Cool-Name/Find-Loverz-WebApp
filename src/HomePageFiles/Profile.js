import React from 'react';
import EditProfile from './EditProfile';
import ProfilePreview from './ProfilePreview';
import './Profile.css';
import NavBar from './NavBar';


const Profile = ({ setUser, user }) => {



    return ( 
        <div>
            <div className="profile">
                <EditProfile user={user} setUser={setUser}></EditProfile>
                <NavBar user={user} setUser={setUser} />
                <div style={{width: "51vw", height: 0}}></div>
                <ProfilePreview user={user} setUser={setUser}></ProfilePreview>
            </div>
        </div>
    );
}
 
export default Profile;