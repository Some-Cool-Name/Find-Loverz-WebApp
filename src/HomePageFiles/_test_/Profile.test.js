import React from 'react';
import 'regenerator-runtime/runtime';
import { BrowserRouter } from "react-router-dom"
import Profile from '../Profile';
import { render, screen, waitFor, fireEvent, cleanup, waitForElementToBeRemoved } from '@testing-library/react';
import renderer, { act } from 'react-test-renderer';


const user = [
    {
        "username":"begin",
        "name":"Sam",
        "Birthday":"1-9-2000",
        "gender":"Male",
        "Sexuality":"Male",
        "bio":"This is the bio",
        "profile_picture":"https://res.cloudinary.com/dkctv74ue/image/upload/v1633950788/index_g5glv5.jpg",
        "location":"braam"
    }
]

it('renders correctly', () => {
    const tree = renderer.create(<BrowserRouter>
        <Profile user={user} setUser = {null}/>
    </BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('Profile Inputs( name, location, bio, interests, profile picture ) : Empty Fields Testing , Should return error ', async () => {

    // render or load Update profile  screen : now we have Update profile screen displayed
    act( () => {
        render( 
        <BrowserRouter>
            <Profile user={user} setUser = {null}/>
        </BrowserRouter>
        );
    });
    
    const name = screen.getByPlaceholderText('name'); 
    act( () => {
        fireEvent.change(name, {target: {value: ''}});  // this are events that change states
    });
    await waitFor(() => { 
        expect(name.value).toBe('');
    });

    const location = screen.getByPlaceholderText('location');
    act( ()=> {
        fireEvent.change(location, {target: {value: ''}})
    });
    await waitFor(() => { 
        expect(location.value).toBe('');
    });

    const bio = screen.getByPlaceholderText('bio');
    act( ()=> {
        fireEvent.change(bio, {target: {value: ''}})
    });
    await waitFor(() => { 
        expect(bio.value).toBe('');
    });

    const UpdateBtn = screen.getByText('Update');
    act( () => {
        fireEvent.click(UpdateBtn);
    });
    await waitFor(() => { 
        expect(UpdateBtn).toBeTruthy();

    });

});

it('Profile Inputs( name, location, bio, interests, profile picture ) : Correct Inputs Testing , Should not return error ', async () => {

    // render or load Update profile  screen : now we have Update profile screen displayed
    act( () => {
        render( 
        <BrowserRouter>
            <Profile user={user} setUser = {null}/>
        </BrowserRouter>
        );
    });
    
    const name = screen.getByPlaceholderText('name'); 
    act( () => {
        fireEvent.change(name, {target: {value: 'Thando'}});  // this are events that change states
    });
    await waitFor(() => { 
        expect(name.value).toBe('Thando');
    });

    const location = screen.getByPlaceholderText('location');
    act( ()=> {
        fireEvent.change(location, {target: {value: 'Braam'}})
    });
    await waitFor(() => { 
        expect(location.value).toBe('Braam');
    });

    const bio = screen.getByPlaceholderText('bio');
    act( ()=> {
        fireEvent.change(bio, {target: {value: 'This is my bio'}})
    });
    await waitFor(() => { 
        expect(bio.value).toBe('This is my bio');
    });

    const UpdateBtn = screen.getByText('Update');
    act( () => {
        fireEvent.click(UpdateBtn);
    });
    await waitFor(() => { 
        expect(UpdateBtn).toBeTruthy();

    });

});

it('Profile Inputs( interests ) : Correct Inputs Testing , Should not return error ', async () => {

    act( () => {
        render( 
        <BrowserRouter>
            <Profile user={user} setUser = {null}/>
        </BrowserRouter>
        );
    });


    const chooseInterests = screen.getByText('Choose Interests');
    act( () => {
        fireEvent.click(chooseInterests);
    });
    await waitFor(() => { 
        expect(screen.getByText('Choose Five Interests')).toBeTruthy();

    });
    const interest1 = screen.getByText('Photography');
    act( () => {
        fireEvent.click(interest1);
    });

    act( () => {
        fireEvent.click(interest1);
    });

    act( () => {
        fireEvent.click(interest1);
    });
    const interest2 = screen.getByText('Shopping');
    act( () => {
        fireEvent.click(interest2);
    });
    const interest3 = screen.getByText('Baking');
    act( () => {
        fireEvent.click(interest3);
    });
    const interest4 = screen.getByText('Comedy');
    act( () => {
        fireEvent.click(interest4);
    });
    const interest5 = screen.getByText('Cooking');
    act( () => {
        fireEvent.click(interest5);
    });

    const close = screen.getByText('Close')
    act( () => {
        fireEvent.click(close);
    });

    const UpdateBtn = screen.getByText('Update');
    act( () => {
        fireEvent.click(UpdateBtn);
    });
    await waitFor(() => { 
        expect(UpdateBtn).toBeTruthy();

    });
});




