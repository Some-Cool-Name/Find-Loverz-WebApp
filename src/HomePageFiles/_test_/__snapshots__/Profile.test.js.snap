// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="profile"
  >
    <div>
      <div
        className="panel-container-1"
      >
        <div
          className="panel-container-2"
        >
          <div
            className="profile-container"
          >
            <div
              className="profile-name"
            >
              <p
                id="feed-username"
              >
                PROFILE
              </p>
            </div>
          </div>
          <div
            id="logout"
          >
            <i
              class="uil uil-sign-out-alt"
              onClick={[Function]}
            />
          </div>
        </div>
        <br />
        <div>
          <img
            className="profile-image"
            id="feed-image"
            src="https://res.cloudinary.com/dkctv74ue/image/upload/v1633950788/index_g5glv5.jpg"
          />
          <p
            className="editProfile"
          >
            edit profile
          </p>
        </div>
        <div>
          <h2
            className="usernameAge"
          >
            begin ,22
          </h2>
          <hr
            className="divider"
            size="6"
          />
          <br />
          <br />
        </div>
        <dir>
          <div
            className="wrapper"
          >
            <label
              className="fieldDescription"
              htmlFor="username"
            >
               Profile Picture
            </label>
            <img
              id="register-pic"
            />
            <input
              accept="image/*"
              id="fileupload"
              onChange={[Function]}
              type="file"
            />
            <p
              className="lblDiscriber"
            >
              NAME
            </p>
            <input
              align="start"
              className="inputfield"
              onChange={[Function]}
              type="text"
              value=""
            />
            <p
              className="lblDiscriber"
            >
              LOCATION
            </p>
            <input
              align="start"
              className="inputfield"
              readOnly={true}
              style={
                Object {
                  "color": "#00bafa",
                }
              }
              type="text"
              value="braam"
            />
            <p
              className="lblDiscriber"
            >
              BIO
            </p>
            <textarea
              className="bio"
              cols="52"
              onChange={[Function]}
              rows="6"
              type="text"
              value=""
            />
            <textarea
              className="bio"
              cols="52"
              onChange={[Function]}
              rows="6"
              type="text"
              value="This is the bio"
            />
            <p
              className="lblDiscriber"
            >
              Interest
            </p>
            <div
              className="interest"
            >
              <div
                className="interest-element"
                style={
                  Object {
                    "borderColor": "#12c2e9",
                    "color": "#12c2e9",
                  }
                }
                title="interestOne"
              >
                no user
              </div>
              <div
                className="interest-element"
                style={
                  Object {
                    "borderColor": "#c471ed",
                    "color": "#c471ed",
                  }
                }
              >
                no user
              </div>
              <div
                className="interest-element"
                style={
                  Object {
                    "borderColor": "#fc32e8",
                    "color": "#fc32e8",
                  }
                }
              >
                no user
              </div>
              <br />
              <div
                className="interest-element"
                style={
                  Object {
                    "borderColor": "#fc32e8",
                    "color": "#fc32e8",
                  }
                }
              >
                no user
              </div>
              <div
                className="interest-element"
              >
                no user
              </div>
            </div>
            <button
              className="intrestbutton"
              id="interests-button"
              onClick={[Function]}
            >
              Choose Interests
            </button>
                                         
            <hr
              className="line2"
              size="2"
              style={
                Object {
                  "Color": "#6e6e6e",
                }
              }
            />
            <div
              className="btnUpdate"
            >
              <a
                className="button"
                href=""
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "grey",
                    "color": "black",
                    "marginTop": 10,
                    "padding": 5,
                    "textDecoration": "none",
                  }
                }
              >
                <b>
                  Update
                </b>
                 
              </a>
            </div>
          </div>
          <div
            id="interests-background"
          >
            <div
              id="interests-interface"
            >
              <h1>
                Choose Five Interests
              </h1>
              <div
                className="interests-container"
              >
                <div
                  className="interest-element"
                  id="i1"
                >
                  Art
                </div>
                <div
                  className="interest-element"
                  id="i2"
                >
                  Baking
                </div>
                <div
                  className="interest-element"
                  id="i3"
                >
                  Board Games
                </div>
                <div
                  className="interest-element"
                  id="i4"
                >
                  Comedy
                </div>
                <div
                  className="interest-element"
                  id="i5"
                >
                  Cooking
                </div>
                <div
                  className="interest-element"
                  id="i6"
                >
                  Cycling
                </div>
                <div
                  className="interest-element"
                  id="i7"
                >
                  Dancing
                </div>
                <div
                  className="interest-element"
                  id="i8"
                >
                  Fashion
                </div>
                <div
                  className="interest-element"
                  id="i9"
                >
                  Fishing
                </div>
                <div
                  className="interest-element"
                  id="i10"
                >
                  Foodie
                </div>
                <div
                  className="interest-element"
                  id="i11"
                >
                  Football
                </div>
                <div
                  className="interest-element"
                  id="i12"
                >
                  Gamer
                </div>
                <div
                  className="interest-element"
                  id="i13"
                >
                  Gardening
                </div>
                <div
                  className="interest-element"
                  id="i14"
                >
                  Golf
                </div>
                <div
                  className="interest-element"
                  id="i15"
                >
                  Gym
                </div>
                <div
                  className="interest-element"
                  id="i16"
                >
                  Hiking
                </div>
                <div
                  className="interest-element"
                  id="i17"
                >
                  Karaoke
                </div>
                <div
                  className="interest-element"
                  id="i18"
                >
                  Movies
                </div>
                <div
                  className="interest-element"
                  id="i19"
                >
                  Music
                </div>
                <div
                  className="interest-element"
                  id="i20"
                >
                  Netflix
                </div>
                <div
                  className="interest-element"
                  id="i21"
                >
                  Pet Lover
                </div>
                <div
                  className="interest-element"
                  id="i22"
                >
                  Photography
                </div>
                <div
                  className="interest-element"
                  id="i23"
                >
                  Picnicking
                </div>
                <div
                  className="interest-element"
                  id="i24"
                >
                  Politics
                </div>
                <div
                  className="interest-element"
                  id="i25"
                >
                  Reading
                </div>
                <div
                  className="interest-element"
                  id="i26"
                >
                  Shopping
                </div>
                <div
                  className="interest-element"
                  id="i27"
                >
                  Spirituality
                </div>
                <div
                  className="interest-element"
                  id="i28"
                >
                  Social Media
                </div>
                <div
                  className="interest-element"
                  id="i29"
                >
                  Sport
                </div>
                <div
                  className="interest-element"
                  id="i30"
                >
                  Surfing
                </div>
                <div
                  className="interest-element"
                  id="i31"
                >
                  Swimming
                </div>
                <div
                  className="interest-element"
                  id="i32"
                >
                  Travel
                </div>
                <div
                  className="interest-element"
                  id="i33"
                >
                  Vlogging
                </div>
                <div
                  className="interest-element"
                  id="i34"
                >
                  Wine
                </div>
                <div
                  className="interest-element"
                  id="i35"
                >
                  Writing
                </div>
                <div
                  className="interest-element"
                  id="i36"
                >
                  Yoga
                </div>
              </div>
              <button
                id="interests-close"
                onClick={[Function]}
              >
                Close
              </button>
            </div>
          </div>
        </dir>
      </div>
    </div>
    <div
      className="navBarNew"
    >
      <p
        id="findLovers"
      >
        <b>
          FindLoverz   
        </b>
      </p>
      <p
        id="divider"
      >
        |
      </p>
      <a
        href="/profile"
        onClick={[Function]}
      >
        <img
          id="NavBarImage"
          src="https://res.cloudinary.com/dkctv74ue/image/upload/v1633950788/index_g5glv5.jpg"
        />
      </a>
      <a
        href="/feed"
        id="navFeed"
        onClick={[Function]}
      >
         
        <b>
          Feed
        </b>
         
      </a>
      <a
        href="/chat"
        id="navChat"
        onClick={[Function]}
      >
         
        <b>
          Chat
        </b>
         
      </a>
      <a
        href="/"
        id="navLogout"
        onClick={[Function]}
      >
         
        <b>
          Logout
        </b>
         
      </a>
    </div>
    <div
      style={
        Object {
          "height": 0,
          "width": "51vw",
        }
      }
    />
    <div
      style={
        Object {
          "position": "absolute",
          "right": "0px",
          "top": "0px",
          "zIndex": -1,
        }
      }
    >
      <div
        className="right-container-1"
      >
        <div
          className="right-container-2"
        >
          <h2
            style={
              Object {
                "color": "#808080",
                "marginRight": "10px",
              }
            }
          >
            PREVIEW
          </h2>
          <div
            className="card-container"
            id="card"
          >
            <img
              alt=""
              className="card-image"
              src="https://res.cloudinary.com/dkctv74ue/image/upload/v1633950788/index_g5glv5.jpg"
            />
            <h2
              className="usernameAge2"
              title="name"
            >
              Sam
            </h2>
            <p
              className="location"
              title="location"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": "#808080",
                    "marginRight": "10px",
                  }
                }
                version="1.1"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 0c-2.761 0-5 2.239-5 5 0 5 5 11 5 11s5-6 5-11c0-2.761-2.239-5-5-5zM8 8.063c-1.691 0-3.063-1.371-3.063-3.063s1.371-3.063 3.063-3.063 3.063 1.371 3.063 3.063-1.371 3.063-3.063 3.063zM6.063 5c0-1.070 0.867-1.938 1.938-1.938s1.938 0.867 1.938 1.938c0 1.070-0.867 1.938-1.938 1.938s-1.938-0.867-1.938-1.938z"
                />
              </svg>
              braam
            </p>
            <div
              className="interest2"
            >
              <div
                className="interest-element2"
                style={
                  Object {
                    "borderColor": "#12c2e9",
                    "color": "#12c2e9",
                  }
                }
                title="intOne"
              >
                Golf
              </div>
              <div
                className="interest-element2"
                style={
                  Object {
                    "borderColor": "#c471ed",
                    "color": "#c471ed",
                  }
                }
                title="intTwo"
              >
                Gym
              </div>
              <div
                className="interest-element2"
                style={
                  Object {
                    "borderColor": "#fc32e8",
                    "color": "#fc32e8",
                  }
                }
                title="intThree"
              >
                Pet Lover
              </div>
              <br />
              <div
                className="interest-element2"
                style={
                  Object {
                    "borderColor": "#fc32e8",
                    "color": "#fc32e8",
                  }
                }
                title="intFour"
              >
                Spirituality
              </div>
              <div
                className="interest-element2"
                title="intFive"
              >
                Vlogging
              </div>
            </div>
            <div
              className="card-bio"
            >
              <div>
                <p
                  title="previewBio"
                >
                   
                  This is the bio
                   
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
